opa/opa.go

```diff
@@ -165,27 +200,27 @@ func (a *inProcessAuthorizer) Authorize(subject string, groups []string, permiss
 	if err != nil {
 		level.Error(a.logger).Log("msg", "failed to evaluate OPA query", "err", err.Error())
 
-		return http.StatusForbidden, false
+		return http.StatusForbidden, false, ""
 	}
 
 	if len(res) == 0 || len(res[0].Expressions) == 0 || res[0].Expressions[0] == nil {
 		level.Error(a.logger).Log("msg", "received a empty OPA response")
 
-		return http.StatusForbidden, false
+		return http.StatusForbidden, false, ""
 	}
 
 	result, ok := (res[0].Expressions[0].Value).(bool)
 	if !ok {
 		level.Error(a.logger).Log("msg", "received a malformed OPA response")
 
-		return http.StatusForbidden, false
+		return http.StatusForbidden, false, ""
 	}
 
 	if !result {
-		return http.StatusForbidden, result
+		return http.StatusForbidden, result, ""
 	}
 
-	return http.StatusOK, result
+	return http.StatusOK, result, ""
```

Comment:
This seems incomplete to me. The in-process authorizer is a full OPA implementation. There is nothing stopping the OPA policy that is loaded in-process from wanting to return authorization data as well. IMO this should include exactly the same logic above in the rest authorizer.